core/modules/wdb/QueryAction.cc

```diff
@@ -164,6 +167,9 @@ QueryAction::Impl::Impl(QueryActionArg const& a)
       _poisoned(false),
       _sendChannel(a.task->sendChannel),
       _user(a.task->user) {
+    char buf[MAX_NAME_LENGTH];
+    gethostname(buf, MAX_NAME_LENGTH);
+    _hostName = buf;
```

Comment:
Also one should always check system service return codes, such as from gethostname().  What to do with the knowledge of an error in this case depends on the bigger picture of this class.  With the design as it stands there seems to be little alternative but to throw an exception.